test/hooks.on-listen.test.js

```diff
@@ -0,0 +1,614 @@
+'use strict'
+
+const { t, before } = require('tap')
+const Fastify = require('../fastify')
+const fp = require('fastify-plugin')
+const dns = require('dns').promises
+
+let localhost
+
+before(async function () {
+  const lookup = await dns.lookup('localhost')
+  localhost = lookup.address
+})
+
+t.test('localhost onListen should be called in order', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  let order = 0
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 0, '1st called in root')
+    done()
+  })
+
+  fastify.addHook('onListen', function (done) {
+    t.equal(order++, 1, '2nd called in root')
+    done()
+  })
+  fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+t.test('localhost async onListen should be called in order', async t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close.bind(fastify))
+  let order = 0
+
+  fastify.addHook('onListen', async function () {
+    t.equal(order++, 0, '1st async called in root')
+  })
+
+  fastify.addHook('onListen', async function () {
+    t.equal(order++, 1, '2nd async called in root')
+  })
+
+  await fastify.listen({
+    host: 'localhost',
+    port: 0
+  })
+})
+t.test('localhost onListen should manage error in sync', t => {
```

Comment:
What is the purpose of this test?

Reply:
It is supposed to show that Fastify can handle errors thrown by onListen, but through review and recent discussion regarding callbacks and resolve/reject, I believe it is not functioning as intended (i.e. useless). Need to see how those discussions turn out before making changes.